### DISCUSSION TITLE
Error handling in read function

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [MarioCarrion/videos#3] [Q&A] [answered]
Not an issue but rather a question:

How any other error besides `io.EOF` would be handled [here](https://github.com/MarioCarrion/videos/blob/a2d4d2d8efb306f44c23e157d0ad10636230f3de/2021/09/02/01-sync/main.go#L99)?

Thanks.

**Comment by MarioCarrion** [comment, marked as answer]
Hello! 

Great question and also it looks like I totally forgot to check other errors, thanks for calling that out.

I approach this in two different ways:
* Return another channel for the error and use `select` together with the one for reading the values (`errGroupWithErrChannel`)
* Define a new type that saves the err value and check that after the reading channel is closed (`errGroupWithType`)

Both implementations are [here](https://github.com/MarioCarrion/videos/tree/d8ac930ac7f7b9218e0eb00fdddab6254d51f8c3/2021/09/02/01-sync-read); to keep everything simple I'm only showing the example using `errgroup`, but using `sync` should be similar.

I typically start with the function and then depending if I need more details about the read process I implement the type one.

I hope that helps.
